.devcontainer/devcontainer.json

```diff
@@ -10,9 +10,6 @@
     },
     "ghcr.io/devcontainers/features/aws-cli:1": {}
   },
-  "postCreateCommand": [
-    "pip install --user -U pre-commit && pre-commit install-hooks",
-    "npm install -g aws-cdk@2.151.0"
-  ],
+  "postCreateCommand": "./tools/setup.sh",
```

Comment:
i'm not a fan of putting commands in an external file because making changes to the external file will not trigger an update to the container.

Reply:
The list of commands to run became too long to be listed on a single line. It's also possible to add comment to the bash script to provide additional context. In practice, there's barely the need to rebuild the dev container each time that the instructions changes. For the developer who modify the setup instructions, it's faster to rerun the script manually or run only the command that has changed if needed.

Reply:
ok, +1 for comments